### PR TITLE
Add multiple methods to Array

### PR DESCRIPTION
**Pull request by simon0191**
- Bump version to 0.9.0
- Add unshift, shift, dup, to_s and reverse to Array
- Update ruby-sys required version to 0.3.0

I would like to add some tests. I was thinking of creating a ruby project and add rspec tests. What do you think @d-unseductable ?

### CODE REVIEW

**Comment by malept** [comment]
_As a note for anyone looking at this, it requires https://github.com/steveklabnik/ruby-sys/pull/10 to be released._

File: NEWS.md [hunk 1]
```diff
@@ -1,6 +1,16 @@
 # Changes by Version
 
 ## Unreleased
+## [0.9.0] - 2016-10-06
```
Comment (malept) [inline NEWS.md]: That date is probably going to be inaccurate.

**Comment by elifoster** [comment]
Regarding tests, Travis does not support multi language stuff, so if you wanted it done through CI you'd either need a script to install either rust or ruby (I'd go with rust tbh, much simpler) or to use multiple CI services.

**Comment by malept** [comment]
It's not too difficult to run both Rust and Ruby tests with Travis CI, I do so in [`rusty_blank`](https://github.com/malept/rusty_blank/blob/master/.travis.yml) (powered by Ruru).

My personal preference is to have Ruby tests written with minitest because that's what's already bundled with Ruby and it's similar to how Rust tests are written. But it's not my project, so I'll leave it up to @d-unseductable.

**Comment by d-unseductable** [comment]
@elifoster, do you mean the \```ruby sections in documentation?

Those are just examples and they are not executed with CI. Every rust example (doctest) has its corresponding ruby example which is basically the same code written in ruby. IMO it's sometimes easier for users to understand an example reading both rust code and its ruby analog.

@malept, it's a good idea to have rspec/minitest test suite.

Since most of the ruru methods are just simple bindings (receive and process arguments and call the corresponding `ruby-sys` function, like `Array.push()` or `RString::new()`), IMO one test (doctest) per method is enough to be sure that those bindings work correctly.

For more complicated cases I wanted to have integration tests using ruby test frameworks, but at that time it was problematic to set the things up. It was quite a long time ago, before `thermite` was created. So maybe it's good time to try it once again :)

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ruru"
-version = "0.8.1"
+version = "0.9.0"
```
Comment (d-unseductable) [inline Cargo.toml]: Please revert the changes of `Cargo.toml` and rebase the branch on top of `master`. I released a new version of `ruby-sys` and updated this file accordingly.

File: NEWS.md [hunk 3]
```diff
@@ -1,6 +1,16 @@
...
+
+### Added
```
Comment (d-unseductable) [inline NEWS.md]: Please leave these changes under `Unreleased` section for now

File: src/binding/array.rs [hunk 4]
```diff
@@ -25,3 +25,27 @@ pub fn push(array: Value, item: Value) -> Value {
 pub fn store(array: Value, offset: i64, item: Value) -> Value {
     unsafe { array::rb_ary_store(array, offset as c_long, item) }
 }
+
+pub fn pop(array: Value) -> Value {
+    unsafe { array::rb_ary_pop(array) }
+}
+
+pub fn unshift(array: Value, item: Value) -> Value {
+    unsafe { array::rb_ary_unshift(array,item) }
```
Comment (d-unseductable) [inline src/binding/array.rs]: Missing whitespace after `,`

File: src/binding/array.rs [hunk 5]
```diff
@@ -25,3 +25,27 @@ pub fn push(array: Value, item: Value) -> Value {
...
+    unsafe { array::rb_ary_dup(array) }
+}
+
+pub fn to_s(array: Value) -> Value {
+    unsafe { array::rb_ary_to_s(array) }
+}
+
+pub fn reverse(array: Value) -> Value {
+    unsafe { array::rb_ary_reverse(array) }
+}
```
Comment (d-unseductable) [inline src/binding/array.rs]: Missing newline at the end of the file

File: src/class/array.rs [hunk 6]
```diff
@@ -177,6 +177,193 @@ impl Array {
...
+    /// ```
+    /// use ruru::{Array, Fixnum, Object};
+    ///
+    /// let mut array = Array::new().push(Fixnum::new(1));
+    ///
+    /// let mut copy = array.dup();
+    ///
+    /// assert_eq!(
+    ///    array.at(0).try_convert_to::<Fixnum>(),
+    ///    Ok( copy.at(0).try_convert_to::<Fixnum>() ));
```
Comment (d-unseductable) [inline src/class/array.rs]: Remove whitespace after `(` and before `)`

File: src/class/array.rs [hunk 7]
```diff
@@ -177,6 +177,193 @@ impl Array {
...
+    /// # Examples
+    ///
+    /// ```
+    /// use ruru::{Array, Fixnum, Object};
+    ///
+    /// let mut array = Array::new().push(Fixnum::new(1)).push(Fixnum::new(2));
+    /// let string = array.to_s().to_string();
+    ///
+    /// assert_eq!(string, "12".to_string());
+    ///
```
Comment (d-unseductable) [inline src/class/array.rs]: Extra empty line

File: Cargo.toml (hunk 2 above)
Comment (d-unseductable) [inline Cargo.toml]: Oh again I've pushed only release tag without the commit itself 😄 Thanks

**Comment by simon0191** [comment]
👌 Working on:
[ ] Format code
[ ] Remove version
[ ] Adjust NEWS
[ ] Adjust Cargo.toml
[ ] Rebase

**Comment by d-unseductable** [comment]
Great job, @simon0191! Thank you!

**Comment by d-unseductable** [comment]
FYI @simon0191 969ef55f8a926536154faa10071c4679c7ca51b7...8b92dc46c24ce9bad2a4224a3806abd46b340c36
